tests/test_modeling_perceiver.py

```diff
@@ -759,28 +759,32 @@ def test_problem_types(self):
 
     @require_torch_multi_gpu
     def test_multi_gpu_data_parallel_forward(self):
-        for model_class in self.all_model_classes:
-            config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_model_class(model_class)
-
-            # some params shouldn't be scattered by nn.DataParallel
-            # so just remove them if they are present.
-            blacklist_non_batched_params = ["head_mask", "decoder_head_mask", "cross_attn_head_mask"]
-            for k in blacklist_non_batched_params:
-                inputs_dict.pop(k, None)
-
-            # move input tensors to cuda:O
-            for k, v in inputs_dict.items():
-                if torch.is_tensor(v):
-                    inputs_dict[k] = v.to(0)
-
-            model = model_class(config=config)
-            model.to(0)
-            model.eval()
+        # This test currently doesn't pass because of a bug in PyTorch: https://github.com/pytorch/pytorch/issues/36035
```

Comment:
@NielsRogge @LysandreJik @sgugger - the reason for this bug is pretty deeply rooted in PyTorch: https://github.com/pytorch/pytorch/issues/36035

The PyTorch author's recommend switching to DDP here which we already have done for training. I think we should do the same for the multi-gpu tests here no? What do you think?

Reply:
Yes, Accelerate also uses DDP by default, right?

Can we, instead of adding a comment, make it a best practice to use the `@unittest.skip("reason=")` annotator when skipping a test?

Reply:
We should ideally test both `DataParallel` and `DistributedDataParallel`. If we have common tests for both with proper attributes, we can them have an automated script extract that info and have a table ready for the users that tell them which model works with DP and which one works for DDP.

In the meantime, there should also be a clear mention on the doc page of Perceiver that it does not work with DataParallel.

Reply:
Agree - I'll take care of this first thing on Monday